### DISCUSSION TITLE
MavenProject.getDependencyArtifacts()[0].getFile()

### DISCUSSION THREAD

**Discussion by kwin** [eclipse-m2e/m2e-core#1729] [Q&A]
In a mojo I evaluate the dependency artifact's files via `MavenProject.getDependencyArtifacts()[0].getFile()`. For dependencies which are Eclipse projects this sometimes returns

1. The path to the project's pom.xml and sometimes
2. The path to the `target/classes` folder for dependencies which are Eclipse projects

While I do understand the latter I am wondering why the path to the pom.xml is returned (for dependencies with type != pom). Is that a bug or a feature?

**Comment by kwin** [comment]
The logic seems to be in https://github.com/eclipse-m2e/m2e-core/blob/5f945212542f8aee7ca9c20dd59a3dd2dfee3d5a/org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/EclipseWorkspaceArtifactRepository.java#L84-L102. But to me returning the path to the `pom.xml` should never happen if extension is not `pom` but in fact this seems to be the fallback in case no better directory can be determined.

**Reply by kwin** [reply]
I opened https://github.com/eclipse-m2e/m2e-core/issues/1730 to improve the situation.

**Comment by laeubi** [comment]
> While I do understand the latter I am wondering why the path to the pom.xml is returned (for dependencies with type != pom). Is that a bug or a feature?

To "resolve" a workspace project maven requires an existing file. So if the project has no target/classes an arbitrary exiting file has to be returned.
